dcm4che-core/src/main/java/org/dcm4che3/data/BulkData.java

```diff
@@ -333,17 +348,11 @@ public String uriWithoutQuery() {
         return uri.substring(0, uriPathEnd);
     }
 
-    public InputStream openStream() throws IOException {
+    public ImageInputStream openImageInputStream() throws IOException {
         if (uri == null)
             throw new IllegalStateException("uri: null");
- 
-        if (!uri.startsWith("file:"))
-            return new URL(uri).openStream();
-
-        InputStream in = new FileInputStream(getFile());
-        StreamUtils.skipFully(in, offset);
-        return in;
 
+        return this.uriLoader.openStream(this.toFileURI());
```

Comment:
The details of how to open or reuse an image input stream is now assigned to the ImageInputStreamLoader.  This allows deployments to override the IIS behaviour for different types including vendor specific URI schemes.